sklearn/decomposition/factor_analysis.py

```diff
@@ -347,3 +359,38 @@ def score(self, X, y=None):
             Average log-likelihood of the samples under the current model
         """
         return np.mean(self.score_samples(X))
+
+    def _rotate(self, method="varimax", n_components=None, tol=1e-6):
+        "Rotate the factor analysis solution."
+        implemented = {"varimax", "quartimax"}
+        if method in implemented:
+            self.components_ = self._ortho_rotation(
+                self.components_.T, method=method, eps=tol)[:n_components]
+        else:
+            raise NotImplementedError("'method' must be in %s, not %s"
+                                      % (implemented, method))
+
+    def _ortho_rotation(self, components, method='varimax', eps=1e-6,
+                        itermax=100):
+        """Return rotated components."""
+        if (method == 'varimax'):
+            gamma = 1.0
+        elif (method == 'quartimax'):
+            gamma = 0.0
+
+        nrow, ncol = components.shape
+        rotation_matrix = np.eye(ncol)
+        var = 0
+
+        for _ in range(itermax):
+            comp_rot = np.dot(components, rotation_matrix)
+            tmp = np.diag((comp_rot ** 2).sum(axis=0)) / nrow * gamma
```

Comment:
why do all this computation unnecessarily in the quartimax case?

Can't we just use `if` here?